### PR TITLE
New account domain

### PR DESCRIPTION
**Pull request by orangejenny**
Reopening https://github.com/dimagi/commcare-hq/pull/7310 to merge with master.

@sravfeyn addressed your comments from the previous PR

cc @amsagoff

### CODE REVIEW

**Comment by orangejenny** [comment]
QA ticket: http://manage.dimagi.com/default.asp?175221

**Comment by czue** [comment]
@orangejenny no big deal but when you get a chance you should read about `git rebase -i` which will allow you to clear up your commit history to avoid the "todo" and "wip" style stuff.

**Comment by orangejenny** [comment]
@czue sure, will do

**Comment by sravfeyn** [comment]
Looks good. Waiting for QA to complete. Side note - having a linter on your IDE could save you from dimagimon.

File: corehq/apps/hqwebapp/views.py [hunk 1]
```diff
@@ -309,33 +309,35 @@ def csrf_failure(request, reason=None, template_name="csrf_failure.html"):
...
         context.update({
-            'domain': domain,
+            'domain': domain_name,
+            'hr_name': domain.hr_name if domain else domain_name,
             'next': req.REQUEST.get('next', '/a/%s/' % domain),
         })
 
     return django_login(req, template_name=template_name,
-                        authentication_form=EmailAuthenticationForm if not domain else CloudCareAuthenticationForm,
+                        authentication_form=EmailAuthenticationForm if not domain_name else CloudCareAuthenticationForm,
```
Comment (dimagimon) [inline corehq/apps/hqwebapp/views.py]: E501 line too long (120 > 115 characters)

**Comment by orangejenny** [comment]
@sravfeyn QA passed, should be good to go. I added some commits during QA, might want to glance over those.

**Pull request merged**